code/_helpers/icons.dm

```diff
@@ -828,6 +828,21 @@ The _flatIcons list is a cache for generated icon files.
 	flat_icon.AddAlphaMask(alpha_mask)//Finally, let's mix in a distortion effect.
 	return flat_icon
 
+/proc/build_disappear_icon(atom/A)
```

Comment:
Please cache the result of these icon procs.

Reply:
Why do we need to cache it if it used only one time and exclusive for signle atom?

Reply:
And how I can cache it?

Reply:
You need to cache it because icon procs are one of the costliest things in the game and using them shouldn't be done for disposable stuff.
I'm not sure how you would cache this. In this instance I would say you should use an alpha mask filter or not do it at all. The GetFlatIcon()-based animation here is almost certainly not going to get merged.